src/main/java/org/gooru/insights/services/BaseESServiceImpl.java

```diff
@@ -414,21 +428,18 @@ private void buildBuckets(String traceId,String index, RequestParamsDTO requestP
 					termBuilder = AggregationBuilders.terms(groupBy[i]).field(fieldName);
 				}
 				if (i == groupBy.length - 1) {
-					bucketAggregation(index, requestParamsDTO, termBuilder, metricsName);
+					includeMetricsAggregation(index, requestParamsDTO, filterAggregationBuilder, searchRequestBuilder, termBuilder, null, null, metricsName);
 					includeOrder(requestParamsDTO, validatedData, groupBy[i], termBuilder, null, metricsName);
 					termBuilder.size(0);
 				}
 			}
-			if (getBaseAPIService().checkNull(requestParamsDTO.getFilter())) {
-				FilterAggregationBuilder filterBuilder = null;
-				if (filterBuilder == null) {
-					filterBuilder = includeFilterAggregate(index, requestParamsDTO.getFilter(),validatedData, filterData, userFilter);
-				}
+				
+			if(filterAggregationBuilder  != null) {
 				if (termBuilder != null) {
```

Comment:
@RenukaGurumurthy if and else both condition will create the termBuilder object, so remove the not null statement.